src/pages/ContentfulPage.vue

```diff
@@ -214,34 +236,57 @@ export default {
 				contentKey: this.$route?.meta?.contentfulPage(this.$route)?.trim(),
 			};
 		},
-		preFetch(config, client, args) {
-			return client.query({
-				query: contentfulEntries,
-				variables: {
-					contentType: 'page',
-					contentKey: args?.route?.meta?.contentfulPage(args?.route)?.trim(),
+		async preFetch(config, client, args) {
+			if (args?.route?.path === '/lp/home-ml') {
+				await client.query({ query: pageQuery });
+				const result = await client.query({ query: experimentQuery, variables: { id: manualLendingLPExpKey } });
+				const version = result?.data?.experiment?.version;
+				const { enabled } = getExperimentSettingCached(client, manualLendingLPExpKey);
+				if (enabled) {
+					trackExperimentVersion(
+						client,
+						this.$kvTrackEvent,
+						'Paid home',
+						manualLendingLPExpKey,
+						'EXP-MARS-124-May2022'
+					);
+					if (version === 'b') {
+						return Promise.reject({
+							path: '/lp/home-mlv',
+							query: args?.route?.query,
+							hash: args?.route?.hash,
+						});
+					}
 				}
-			}).then(({ data }) => {
-				// Get Contentful page data
-				const pageData = getPageData(data);
-				if (pageData.error) {
-					// Only import the error page if there is a contentful error
-					return Promise.all([ErrorPage()]);
-				}
-				// Get page frame component
-				const pageFrame = getPageFrameFromType(pageData?.page?.pageType);
-				// Get components for content groups
-				const contentGroups = getContentGroups(pageData);
-				// Start importing all components
-				return Promise.all([
-					pageFrame(),
-					...contentGroups.map(g => g.component()),
-				]);
-			}).then(resolvedImports => {
-				// Call preFetch for page frame and content group components
-				const components = resolvedImports.map(resolvedImport => resolvedImport.default);
-				return preFetchAll(components, client, args);
-			});
+			} else {
```

Comment:
Now that I'm looking at it, I don't think this needs to be wrapped in an `else`. The `return Promise.reject` statement above will handle the case where a redirect is needed, and everything else should fall through to making the contetnfulEntries query. Sorry for the back-and-forth on this!

Reply:
No problem! Didn't realize that either, thanks